tests/Functional/ContentRenderTestBase.php

```diff
@@ -205,4 +206,118 @@ protected function createContactEntity(string $name, string $bundle, int $status
+    $contact_name = $element->findAll('css', 'h3');
+    $this->assertCount(1, $contact_name);
+    $this->assertEquals($name, $contact_name[0]->getText());
+
+    $contact_body = $element->findAll('css', '.ecl-editor');
+    $this->assertCount(1, $contact_body);
+    $this->assertEquals("Body text $name", $contact_body[0]->getText());
+
+    $contacts_html = $element->getHtml();
+    $field_list_assert = new FieldListAssert();
+    $contact_expected_values = [
+      'items' => [
+        [
+          'label' => 'Organisation',
+          'body' => "Organisation $name",
+        ], [
+          'label' => 'Website',
+          'body' => "http://www.example.com/website_$name",
+        ], [
+          'label' => 'Email',
+          'body' => "$name@example.com",
+        ], [
+          'label' => 'Phone number',
+          'body' => "Phone number $name",
+        ], [
+          'label' => 'Mobile number',
+          'body' => "Mobile number $name",
+        ], [
+          'label' => 'Fax number',
+          'body' => "Fax number $name",
+        ], [
+          'label' => 'Postal address',
+          'body' => "Address $name, 1001 Brussels, Belgium",
+        ], [
+          'label' => 'Office',
+          'body' => "Office $name",
+        ], [
+          'label' => 'Social media',
+          'body' => html_entity_decode('&nbsp;') . 'Social media ' . $name,
+        ],
+      ],
+    ];
+    $field_list_assert->assertPattern($contact_expected_values, $contacts_html);
+    $field_list_assert->assertVariant('horizontal', $contacts_html);
+
+    // Assert Press contacts.
+    $press = $element->find('css', '.ecl-u-border-top.ecl-u-border-bottom.ecl-u-border-color-grey-15.ecl-u-mt-s.ecl-u-pt-l.ecl-u-pb-l');
+    $this->assertLinkIcon($press, 'Press contacts', "http://www.example.com/press_contact_$name");
+
+    // Assert contacts Image.
+    $this->assertFeaturedMediaField($element, $name);
+  }
+
+  /**
+   * Asserts field group header.
+   *
+   * @param \Behat\Mink\Element\NodeElement $element
+   *   Field group content.
+   * @param string $title
+   *   Expected title.
+   * @param string $id
+   *   Expected id.
+   */
+  protected function assertContentHeader(NodeElement $element, string $title, string $id = ''): void {
+    $header = $element->find('css', 'h2.ecl-u-type-heading-2');
+    $this->assertEquals($title, $header->getText());
+    if (!empty($id)) {
+      $this->assertEquals($id, $header->getAttribute('id'));
+    }
+  }
+
+  /**
+   * Asserts standalone link template with icon.
+   *
+   * @param \Behat\Mink\Element\NodeElement $element
+   *   Rendered element.
+   * @param string $title
+   *   Link title.
+   * @param string $href
+   *   Link URL.
+   * @param bool $is_external
+   *   Defines whether it is extrernal link or internal.
+   */
+  protected function assertLinkIcon(NodeElement $element, string $title, string $href, bool $is_external = TRUE): void {
```

Comment:
This to me looks like a pattern assertion. Although we don't have a link with icon pattern as such, the final rendered result is very standarized so we may benefit from having it as a pattern assertion. I leave it as nice to have (maybe in a follow up)

Reply:
Since we already have 402 to refactor the tests we can take care of it there.